packages/core/src/template-compiler.ts

```diff
@@ -304,6 +300,11 @@ export default class TemplateCompiler {
     return { syntax, cacheKey };
   }
 
+  @Memoize()
+  private getReversedASTPlugins(ast: unknown[]): unknown[] {
```

Comment:
Two notes:

* we should comment why `reversed` is the correct order
* we should likely considered changing the word from `reversed` to `getOrderedASTPlugins` or simply `astPlugins`